infrastructure/ai/provider_factory.py

```
@@ -70,10 +70,17 @@ class DynamicLLMService(LLMService):
 
     def __init__(self, factory: Optional[LLMProviderFactory] = None):
         self.factory = factory or LLMProviderFactory()
+        self._last_provider: Optional[LLMService] = None
```

Ruff (0.15.12)

[error] Undefined name `Optional`

(F821)